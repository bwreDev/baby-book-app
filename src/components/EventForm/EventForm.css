.event-button {
  background-color: #9e9c67;
  border-radius: 15px;
  padding: 20px;
  border: none;
  text-align: center;
  text-decoration: none;
  display: inline-block;
  font-size: 16px;
  margin: 10px;
  width: 250px;
}

select {
  display: block;
  border: none;
  font-size: 16px;
  font-weight: 700;
  line-height: 1.3;
  padding: 0.6em 1.4em 0.5em 0.8em;
  width: 75%;
  max-width: 75%;
  margin: 15px auto 15px auto;
  border-radius: 0.5em;
  background-color: #fff;
}

select option {
  font-weight: normal;
}

fieldset {
  border: none;
}

input {
  height: 40px;
  border-radius: 15px;
  border-style: none;
  margin: 5px;
  width: 75%;
}

textarea {
  height: 60px;
  border-radius: 15px;
  border-style: none;
  margin: 5px;
  width: 75%;
  font-size: 16px;
}

@media only screen and (min-width: 600px) {
  input {
    margin-top: 20px;
    margin-bottom: 20px;
    height: 60px;
    font-size: 20px;
  }

  select {
    font-size: 20px;
    font-weight: normal;
  }

  textarea {
    height: 80px;
    font-size: 20px;
  }
}

@media only screen and (min-width: 1200px) {
  input {
    margin-top: 30px;
    margin-bottom: 30px;
    height: 80px;
    font-size: 24px;
  }

  select {
    font-size: 24px;
    font-weight: normal;
  }

  textarea {
    height: 100px;
    font-size: 24px;
  }
}
