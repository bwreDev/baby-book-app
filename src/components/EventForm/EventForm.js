import React, { Component } from 'react';
import EventsContext from '../../contexts/EventsContext';
import BabyBookApiService from '../../services/baby-book-api-service';
import Navbar from '../../components/Navbar/Navbar';
import moment from 'moment';
import './EventForm.css';

export default class EventForm extends Component {
  static contextType = EventsContext;

  state = {
    event_type: null,
  };

  onChange = (event) => {
    this.setState({
      event_type: event.target.value,
    });
  };

  handleSubmit = (event) => {
    event.preventDefault();

    let eventObject = {
      title: this.state.event_type,
    };

    if (eventObject.title === 'diaper_changes') {
      eventObject.content = event.target.diaper_changes.value;
    }
    if (eventObject.title === 'feedings') {
      eventObject.content =
        event.target.feedings.value +
        ' ' +
        event.target.feeding_amount.value +
        ' ' +
        event.target.measurement.value;
    }
    if (eventObject.title === 'appointments') {
      eventObject.content =
        event.target.appointments.value +
        ' on - ' +
        moment(event.target.appointments_date.value).format(
          'MMM Do YY'
        );
    }
    if (eventObject.title === 'stretches') {
      eventObject.content = event.target.stretches.value;
    }

    BabyBookApiService.postEvent(eventObject)
      .then(this.context.addEvent)
      .then(this.props.history.push('/user'))
      .catch(this.context.setError);
  };

  render() {
    return (
      <>
        <Navbar />
        <h2>Please select an event to create!</h2>
        <form className='event-form' onSubmit={this.handleSubmit}>
          <h3>Event type</h3>
          <select onChange={this.onChange} name='event_type'>
            <option value=''></option>
            <option value='diaper_changes'>Diaper Changes</option>
            <option value='feedings'>Feedings</option>
            <option value='appointments'>Appointments</option>
            <option value='stretches'>Stretches</option>
          </select>
          {this.state.event_type === 'diaper_changes' ? (
            <fieldset className='diaper_changes'>
              <label htmlFor='diaper_changes'>
                <h3>Diaper changes</h3>
              </label>
              <h4>Option</h4>
              <select name='diaper_changes'>
                <option value=''></option>
                <option value='wet'>Wet</option>
                <option value='dirty'>Dirty</option>
                <option value='mixed'>Mixed</option>
              </select>
            </fieldset>
          ) : null}
          {this.state.event_type === 'feedings' ? (
            <fieldset className='feedings'>
              <label htmlFor='feedings'>
                <h3>Feedings</h3>
              </label>
              <h4>Type</h4>
              <select name='feedings'>
                <option value=''></option>
                <option value='nursing'>Nursing</option>
                <option value='pumped'>Pumped</option>
                <option value='formula'>Formula</option>
              </select>
              <h4>Amount</h4>
              <input
                placeholder='0'
                type='number'
                name='feeding_amount'
              />
              <h4>Unit</h4>
              <select name='measurement'>
                <option value=''></option>
                <option value='ml'>milliliters</option>
                <option value='oz'>ounces</option>
              </select>
            </fieldset>
          ) : null}
          {this.state.event_type === 'appointments' ? (
            <fieldset className='appointments'>
              <label htmlFor='appointments'>
                <h3>Appointment</h3>
              </label>
              <textarea
                placeholder='Type of appointment'
                name='appointments'
              ></textarea>
              <input type='date' name='appointments_date' />
            </fieldset>
          ) : null}
          {this.state.event_type === 'stretches' ? (
            <fieldset className='stretches'>
              <label htmlFor='stretches'>
                <h3>Stretches</h3>
              </label>
              <textarea
                placeholder='Type of stretch'
                name='stretches'
              ></textarea>
            </fieldset>
          ) : null}
          <button className='event-button' type='submit'>
            Save
          </button>
        </form>
      </>
    );
  }
}
