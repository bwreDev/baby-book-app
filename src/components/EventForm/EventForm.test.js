import React from 'react';
import ReactDOM from 'react-dom';
import EventForm from './EventForm';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

it('should render without error', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <MemoryRouter>
      <EventForm />
    </MemoryRouter>,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <EventForm />
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
