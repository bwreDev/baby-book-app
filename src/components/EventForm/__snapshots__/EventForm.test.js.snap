// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
Array [
  <nav
    className="navbar"
    role="navigation"
  >
    <ul
      className="navbar-list"
    >
      <li>
        <a
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="/register"
          onClick={[Function]}
        >
          Register
        </a>
      </li>
      <li>
        <a
          href="/login"
          onClick={[Function]}
        >
          Log in
        </a>
      </li>
    </ul>
  </nav>,
  <form
    className="event-form"
    onSubmit={[Function]}
  >
    <select
      name="event_type"
      onChange={[Function]}
    >
      <option
        value=""
      />
      <option
        value="diaper_changes"
      >
        Diaper Changes
      </option>
      <option
        value="feedings"
      >
        Feedings
      </option>
      <option
        value="appointments"
      >
        Appointments
      </option>
      <option
        value="stretches"
      >
        Stretches
      </option>
    </select>
    <button
      className="event-button"
      type="submit"
    >
      Save
    </button>
  </form>,
]
`;
